File: backend/Models/Session.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    public class Session
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string DoctorId { get; set; } = null!;
        public ApplicationUser Doctor { get; set; } = null!;

        [Required]
        public DateTime Date { get; set; }

        [Required]
        public TimeSpan StartTime { get; set; }

        [Required]
        public TimeSpan EndTime { get; set; }

        [Required]
        public double SessionFee { get; set; }

        [Required]
        public string Description { get; set; } = string.Empty;

        [Required]
        public int Capacity { get; set; }

        public bool Canceled { get; set; } = false;

        [Column("ongoing")]
        public bool Ongoing { get; set; } = false;

        public ICollection<Booking> Bookings { get; set; } = new List<Booking>();
    }
}

File: frontend/src/types/Session.ts
import { Booking } from './Booking';

export interface Session {
    id: number;
    doctorId: string;
    doctor?: {
        id?: string;
        firstName?: string;
        lastName?: string;
        email?: string;
    };
    canceled: boolean;
    ongoing: boolean;
    date: string; // ISO date string
    startTime: string; // TimeSpan from C# will be serialized as string (e.g., "14:30:00")
    endTime: string; // TimeSpan from C# will be serialized as string (e.g., "16:30:00")
    sessionFee: number;
    description: string;
    capacity: number;
    bookings?: Booking[];
    // Keep patients for backward compatibility (derived from bookings)
    patients?: Array<{
        id?: string;
        firstName?: string;
        lastName?: string;
        email?: string;
        contactNumbers?: string[];
    }>;
}